lib/graphql/account_settings/cloud_object/address_cloud_object.dart

```diff
@@ -0,0 +1,99 @@
+import 'dart:convert';
+
+import 'package:autonomy_flutter/database/entity/wallet_address.dart';
+import 'package:autonomy_flutter/graphql/account_settings/account_settings_db.dart';
+
+class WalletAddressCloudObject {
+  final AccountSettingsDB _accountSettingsDB;
+
+  WalletAddressCloudObject(this._accountSettingsDB);
+
+  Future<void> deleteAddress(WalletAddress address) async {
+    await _accountSettingsDB.delete([address.key]);
+  }
+
+  Future<void> deleteAddressesByUuid(String uuid) async {
+    final addressesWithUUid = findByWalletID(uuid);
+    await _accountSettingsDB
+        .delete(addressesWithUUid.map((e) => e.key).toList());
+  }
+
+  List<WalletAddress> findAddressesWithHiddenStatus(bool isHidden) {
+    final allAddresses = getAllAddresses();
+    return allAddresses
+        .where((element) => element.isHidden == isHidden)
+        .toList();
+  }
+
+  WalletAddress? findByAddress(String address) {
+    final value = _accountSettingsDB.query([address]);
```

Comment:
please note that address is key

Reply:
it is defined in WalletAddress definition

Reply:
NO i mean that the `query` method request list of `key`, not list of `address`.

Reply:
from the purpose of this service,  it cares about addresses, not key. 
when you use it, you want to get wallet of some addresses, not some key

Reply:
`AccountSettingsDB` use key to query.
`WalletAddressCloudObject` use address to find wallet, even if address isn't key, we still have to do so, because that what you want from this service

Reply:
if you suppose key is address, please add a comment, else you have to getAllWallet then filter by address.

Reply:
i can add comments to make it more clear.
but other way to notice that is here
```
class WalletAddress implements SettingObject {
  @override
  String get key => address;